src/test/java/com/awesomeorg/airlineservice/integration_tests/ReservationControllerIntegrationTest.java

```diff
@@ -0,0 +1,123 @@
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ReservationService reservationService;
+
+    @Autowired
+    private PassengerService passengerService;
+
+    @Autowired
+    private TicketService ticketService;
+
+
+    @Test
+    @SneakyThrows
+    @DirtiesContext
+    public void createReservationTest() {
+
+        CreatePassengerRequest passengerRequest = new CreatePassengerRequest();
+        passengerRequest.setFirstName("Angela");
+        passengerRequest.setLastName("Smith");
+        passengerRequest.setPhoneNumber("987-654-3210");
+        passengerRequest.setEmailAddress("a.smith@example.com");
+
+        Passenger passenger = passengerService.createPassenger(passengerRequest);
+
+
+        TicketQuery ticketRequest = new TicketQuery();
+        ticketRequest.setDateOfPurchase(LocalDate.parse("2023-01-15"));
+        ticketRequest.setDateOfFlight(LocalDate.parse("2024-03-16"));
+        ticketRequest.setDateOfReturn(LocalDate.parse("2024-05-17"));
+        ticketRequest.setSeat(5);
+        ticketRequest.setPriceOfTicket((int) 1500.60);
+
+        Ticket ticket = ticketService.createTicket(ticketRequest);
+
+
+        CreateReservationRequest reservationRequest = new CreateReservationRequest();
+        reservationRequest.setFlightNumber("RKH987");
+        reservationRequest.setDepartureAirport("KOR");
+        reservationRequest.setArrivalAirport("EST");
+        reservationRequest.setDepartureTime(LocalDateTime.parse("2025-03-25T08:00:00"));
+        reservationRequest.setArrivalTime(LocalDateTime.parse("2025-03-25T12:00:00"));
+        reservationRequest.setTripType("Round-trip");
+        reservationRequest.setDeparture("Korea");
+        reservationRequest.setDestination("Estonia");
+        reservationRequest.setNumberOfCustomerSeats(1);
+        reservationRequest.setClassOfFlight(Reservation.FlightClass.BUSINESS);
+        reservationRequest.setDepartureDate(LocalDate.parse("2025-03-25"));
+        reservationRequest.setReturnDate(LocalDate.parse("2025-05-25"));
+        reservationRequest.setReservationStatus(Reservation.Status.CONFIRMED);
+        reservationRequest.setTicketId(10L);
+        reservationRequest.setPassengerId(8L);
+
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/reservations")
+                        .header("passenger-id", passenger.getId())
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\"flightNumber\":\"RKH987\",\"departureAirport\":\"KOR\",\"arrivalAirport\":\"EST\",\"departureTime\":\"2025-03-25T08:00:00\",\"arrivalTime\":\"2025-03-25T12:00:00\",\"tripType\":\"Round-trip\",\"departure\":\"Korea\",\"destination\":\"Estonia\",\"numberOfCustomerSeats\":1,\"classOfFlight\":\"BUSINESS\",\"departureDate\":\"2025-03-25\",\"returnDate\":\"2025-05-25\",\"reservationStatus\":\"CONFIRMED\",\"ticketId\":" + ticket.getId() + ",\"passengerId\":" + passenger.getId() + "}"))
```

Comment:
Also why this? you have mapper, use it to convert java object into the json string